src/test/java/com/example/gildedrose/service/ItemUpdaterTest.java

```diff
@@ -20,4 +22,14 @@ public void updatesTheItemRepository() {
         itemUpdater.performUpdate();
         verify(itemRepository, times(1)).update(new ArrayList<>());
     }
+
+    @SpyBean
+    private ItemUpdater itemUpdater;
+
+    @Test
+    public void updatesTheRepositoryWhenScheduled() {
+        await()
+                .atMost(Duration.FIVE_SECONDS)
```

Comment:
**suggestion** If you make the cron expression configurable, you should be able to use `@SprintBootTest` to configure this test with a value that is suitable for testing, like every 5s.

The `properties` argument to `@SpringBootTest` could come in handy depending on how you choose to make the cron expression configurable. 

```java
@SpringBootTest(properties = { "gilded-rose.update-item-schedule=..." })
```
There are other ways to configure this in test as well. [Here](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-testing) are the docs for Spring Boot Testing in case you don't have a link handy. (I find it really hard to find the documentation I want when it comes to Spring 😅 ).

Lastly, if you decide to move the scheduling into a dedicated component, I think there might be a legitimate case for not testing the schedule. It would come down to how much value do we get based on the cost of the test.